### PR TITLE
Final part of PhysX/Whitebox/NvCloth/Blast Standarization of paths and multiple fixes

### PR DESCRIPTION
**Pull request by AMZN-AlexOteiza**
Periodic suite, Main suite and optimized test suite are passing for all tests of every category.
Final folder structure:
- **root folder:** Contains the Test suites
image

- **utils**: Utility files **only for** test suites(not usual), such as file management 

- **tests**: Contains the tests, with optionally subfolders for categories.
Tests have Feature_TestThatDoesSomething
Subfolders must use snake_camelcase
Ex: Physics
image
Ex: WhiteBox
image
Ex: NvCloth
image

Non-test files as utils must not have Feature_ format, instead UpperCamelCase:
Ex: Blast
image

Apart from removing the deprecated CXXXX prefix, names have been rewritten in most cases for better clarity
Physics Renames (Including names of levels that didn't match the test name + ScriptCanvas + Regset files)
```
C14654881_CharacterController_SwitchLevels                     ->  CharacterController_SwitchLevels 
C4044695_PhysXCollider_AddMultipleSurfaceFbx                   ->  Collider_MultipleSurfaceOptions 
C4976236_AddPhysxColliderComponent                             ->  Collider_AddColliderComponent 
C4976242_Collision_SameCollisionlayerSameCollisiongroup        ->  Collider_SameCollisionGroupSameLayerCollide 
C4976243_Collision_SameCollisionGroupDiffCollisionLayers       ->  Collider_SameCollisionGroupDiffLayersCollide 
C4976244_Collider_SameGroupSameLayerCollision                  ->  Collider_SameCollisionGroupSameCustomLayerCollide 
C4976245_PhysXCollider_CollisionLayerTest                      ->  Collider_NoneCollisionGroupSameLayerNotCollide 
C4982593_PhysXCollider_CollisionLayerTest                      ->  Collider_DiffCollisionGroupDiffCollidingLayersNotCollide 
C4982595_Collider_TriggerDisablesCollision                     ->  Collider_TriggerPassThrough 
C4982797_Collider_ColliderOffset                               ->  Collider_ColliderPositionOffset 
C4982798_Collider_ColliderRotationOffset                       ->  Collider_ColliderRotationOffset 
C4982800_PhysXColliderShape_CanBeSelected                      ->  Collider_SphereShapeEditting 
C4982801_PhysXColliderShape_CanBeSelected                      ->  Collider_BoxShapeEditting 
C4982802_PhysXColliderShape_CanBeSelected                      ->  Collider_CapsuleShapeEditting 
C4982803_Enable_PxMesh_Option                                  ->  Collider_PxMeshConvexMeshCollides 
C14861498_ConfirmError_NoPxMesh                                ->  Collider_PxMeshErrorIfNoMesh 
C14861500_DefaultSetting_ColliderShape                         ->  Collider_CheckDefaultShapeSettingIsPxMesh 
C14861501_PhysXCollider_RenderMeshAutoAssigned                 ->  Collider_PxMeshAutoAssignedWhenAddingRenderMeshComponent 
C14861502_PhysXCollider_AssetAutoAssigned                      ->  Collider_PxMeshAutoAssignedWhenModifyingRenderMeshComponent 
C14861504_RenderMeshAsset_WithNoPxAsset                        ->  Collider_PxMeshNotAutoAssignedWhenRenderMeshNoPhysicsFbx 
C19578018_ShapeColliderWithNoShapeComponent                    ->  ShapeCollider_InactiveWhenNoShapeComponent 
C19578021_ShapeCollider_CanBeAdded                             ->  ShapeCollider_CanBeAddedWitNoWarnings 
C19723164_ShapeColliders_WontCrashEditor                       ->  ShapeCollider_LargeNumberOfShapeCollidersWontCrashEditor 
C24308873_CylinderShapeCollider_CollidesWithPhysXTerrain       ->  ShapeCollider_CylinderShapeCollides 
Collider_CollisionGroups                                       ->  Collider_CollisionGroupsWorkflow 
C3510644_Collider_CollisionGroups                              ->  Collider_CollisionGroupsWorkflow 
Collider_NewGroup                                              ->  Collider_AddingNewGroupWorks 
C4976227_Collider_NewGroup                                     ->  Collider_AddingNewGroupWorks 
C5932040_ForceRegion_CubeExertsWorldForce                      ->  ForceRegion_WorldSpaceForceOnRigidBodies 
C5932041_PhysXForceRegion_LocalSpaceForceOnRigidBodies         ->  ForceRegion_LocalSpaceForceOnRigidBodies 
C5932042_PhysXForceRegion_LinearDamping                        ->  ForceRegion_LinearDampingForceOnRigidBodies 
C5932043_ForceRegion_SimpleDragOnRigidBodies                   ->  ForceRegion_SimpleDragForceOnRigidBodies 
C5932043_ForceRegion_SimpleDragOnRigidBody                     ->  ForceRegion_SimpleDragForceOnRigidBodies 
C5932044_ForceRegion_PointForceOnRigidBody                     ->  ForceRegion_PointForceOnRigidBodies 
C5932044_PhysX_ForceRegion_PointForceOnRigidBodies             ->  ForceRegion_PointForceOnRigidBodies 
C5932045_ForceRegion_Spline                                    ->  ForceRegion_SplineForceOnRigidBodies 
C5959759_RigidBody_ForceRegionSpherePointForce                 ->  ForceRegion_SphereShapedForce 
C5959760_PhysXForceRegion_PointForceExertion                   ->  ForceRegion_CapsuleShapedForce 
C5959761_ForceRegion_PhysAssetExertsPointForce                 ->  ForceRegion_PxMeshShapedForce 
C5959761_ForceRegion_ExertsPointForce_Sedan                    ->  ForceRegion_PxMeshShapedForce 
C5959763_ForceRegion_ForceRegionImpulsesCube                   ->  ForceRegion_ImpulsesBoxShapedRigidBody 
C5959764_ForceRegion_ForceRegionImpulsesCapsule                ->  ForceRegion_ImpulsesCapsuleShapedRigidBody 
C5959765_ForceRegion_AssetGetsImpulsed                         ->  ForceRegion_ImpulsesPxMeshShapedRigidBody 
C5959808_ForceRegion_PositionOffset                            ->  ForceRegion_PositionOffset 
C5959809_ForceRegion_RotationalOffset                          ->  ForceRegion_RotationalOffset 
C5959810_ForceRegion_ForceRegionCombinesForces                 ->  ForceRegion_MultipleForcesInSameComponentCombineForces 
C5968759_ForceRegion_ExertsSeveralForcesOnRigidBody            ->  ForceRegion_MultipleComponentsCombineForces 
C5968760_ForceRegion_CheckNetForceChange                       ->  ForceRegion_MovingForceRegionChangesNetForce 
C6090546_ForceRegion_SliceFileInstantiates                     ->  ForceRegion_SliceFileInstantiates 
C6090547_ForceRegion_ParentChildForceRegions                   ->  ForceRegion_ParentChildForcesCombineForces 
C6090550_ForceRegion_WorldSpaceForceNegative                   ->  ForceRegion_ZeroWorldSpaceForceDoesNothing 
C6090551_ForceRegion_LocalSpaceForceNegative                   ->  ForceRegion_ZeroLocalSpaceForceDoesNothing 
C6090552_ForceRegion_LinearDampingNegative                     ->  ForceRegion_ZeroLinearDampingDoesNothing 
C6090553_ForceRegion_SimpleDragForceOnRigidBodies              ->  ForceRegion_ZeroSimpleDragForceDoesNothing 
C6090554_ForceRegion_PointForceNegative                        ->  ForceRegion_ZeroPointForceDoesNothing 
C6090555_ForceRegion_SplineFollowOnRigidBodies                 ->  ForceRegion_ZeroSplineForceDoesNothing 
C6321601_Force_HighValuesDirectionAxes                         ->  ForceRegion_HighValuesDirectionAxesWorkWithNoError 
C12868578_ForceRegion_DirectionHasNoAffectOnMagnitude          ->  ForceRegion_DirectionHasNoAffectOnTotalForce 
C12868580_ForceRegion_SplineModifiedTransform                  ->  ForceRegion_SplineRegionWithModifiedTransform 
C12905527_ForceRegion_MagnitudeDeviation                       ->  ForceRegion_SmallMagnitudeDeviationOnLargeForces 
C12905528_ForceRegion_WithNonTriggerCollider                   ->  ForceRegion_WithNonTriggerColliderWarning 
C15845879_ForceRegion_HighLinearDampingForce                   ->  ForceRegion_NoQuiverOnHighLinearDampingForce 
C5689529_Verify_Terrain_RigidBody_Collider_Mesh                ->  Physics_VerifyColliderRigidBodyMeshAndTerrainWorkTogether 
C6131473_StaticSlice_OnDynamicSliceSpawn                       ->  Physics_DynamicSliceWithPhysNotSpawnsStaticSlice.py 
C14976307_Gravity_SetGravityWorks                              ->  RigidBody_SetGravityWorks 
C15425929_Undo_Redo                                            ->  Physics_UndoRedoWorksOnEntityWithPhysComponents 
C19536274_GetCollisionName_PrintsName                          ->  ScriptCanvas_GetCollisionNameReturnsName 
C19536274_NameNode_Prints                                      ->  ScriptCanvas_GetCollisionName 
C19536277_GetCollisionName_PrintsNothing                       ->  ScriptCanvas_GetCollisionNameReturnsNothingWhenHasToggledLayer 
C29032500_EditorComponents_WorldBodyBusWorks                   ->  Physics_WorldBodyBusWorksOnEditorComponents 
C18243580_Joints_Fixed2BodiesConstrained                       ->  Joints_Fixed2BodiesConstrained 
C18243581_Joints_FixedBreakable                                ->  Joints_FixedBreakable 
C18243582_Joints_FixedLeadFollowerCollide                      ->  Joints_FixedLeadFollowerCollide 
C18243583_Joints_Hinge2BodiesConstrained                       ->  Joints_Hinge2BodiesConstrained 
C18243584_Joints_HingeSoftLimitsConstrained                    ->  Joints_HingeSoftLimitsConstrained 
C18243585_Joints_HingeNoLimitsConstrained                      ->  Joints_HingeNoLimitsConstrained 
C18243586_Joints_HingeLeadFollowerCollide                      ->  Joints_HingeLeadFollowerCollide 
C18243587_Joints_HingeBreakable                                ->  Joints_HingeBreakable 
C18243588_Joints_Ball2BodiesConstrained                        ->  Joints_Ball2BodiesConstrained 
C18243589_Joints_BallSoftLimitsConstrained                     ->  Joints_BallSoftLimitsConstrained 
C18243590_Joints_BallNoLimitsConstrained                       ->  Joints_BallNoLimitsConstrained 
C18243591_Joints_BallLeadFollowerCollide                       ->  Joints_BallLeadFollowerCollide 
C18243592_Joints_BallBreakable                                 ->  Joints_BallBreakable 
C18243593_Joints_GlobalFrameConstrained                        ->  Joints_GlobalFrameConstrained 
C4044455_Material_libraryChangesInstantly                      ->  Material_LibraryChangesReflectInstantly 
C4044456_Material_FrictionCombine                              ->  Material_FrictionCombine 
C4044457_Material_RestitutionCombine                           ->  Material_RestitutionCombine 
C4044459_Material_DynamicFriction                              ->  Material_DynamicFriction 
C4044460_Material_StaticFriction                               ->  Material_StaticFriction 
C4044461_Material_Restitution                                  ->  Material_Restitution 
C4044694_Material_EmptyLibraryUsesDefault                      ->  Material_EmptyLibraryUsesDefault 
C4044697_Material_PerfaceMaterialValidation                    ->  Material_PerFaceMaterialGetsCorrectMaterial 
C4888315_Material_AddModifyDeleteOnCollider                    ->  Material_LibraryCrudOperationsReflectOnCollider 
C4925577_Materials_MaterialAssignedToTerrain                   ->  Material_CanBeAssignedToTerrain 
C4925577_PhysXMaterials_MaterialAssignedToTerrain              ->  Material_CanBeAssignedToTerrain 
C4925579_Material_AddModifyDeleteOnTerrain                     ->  Material_LibraryCrudOperationsReflectOnTerrain 
C4925580_Material_RagdollBonesMaterial                         ->  Material_RagdollBones 
C4925582_Material_AddModifyDeleteOnRagdollBones                ->  Material_LibraryCrudOperationsReflectOnRagdollBones 
C5296614_PhysXMaterial_ColliderShape                           ->  Material_NoEffectIfNoColliderShape 
C15096732_Material_DefaultLibraryUpdatedAcrossLevels_after     ->  Material_DefaultLibraryUpdatedAcrossLevels_after 
C15096732_Material_DefaultLibraryUpdatedAcrossLevels_before    ->  Material_DefaultLibraryUpdatedAcrossLevels_before 
C15096735_Materials_DefaultLibraryConsistency                  ->  Material_DefaultLibraryConsistentOnAllFeatures 
C15096737_Materials_DefaultMaterialLibraryChanges              ->  Material_DefaultMaterialLibraryChangesWork 
C15096740_Material_LibraryUpdatedCorrectly                     ->  Material_LibraryClearingAssignsDefault 
C15308221_Material_ComponentsInSyncWithLibrary                 ->  Material_ComponentsInSyncWithLibrary 
C15425935_Material_LibraryUpdatedAcrossLevels                  ->  Material_LibraryUpdatedAcrossLevels 
C15556261_PhysXMaterials_CharacterControllerMaterialAssignment ->  Material_CharacterController 
C15563573_Material_AddModifyDeleteOnCharacterController        ->  Material_LibraryCrudOperationsReflectOnCharacterController 
C18977601_Material_FrictionCombinePriority                     ->  Material_FrictionAllCombinePrioritiesWork 
C18981526_Material_RestitutionCombinePriority                  ->  Material_FrictionAllRestitutionPrioritiesWork 
C13895144_Ragdoll_ChangeLevel                                  ->  Ragdoll_LevelSwitchDoesNotCrash 
C13895144_Ragdoll_NoRagdoll                                    ->  Ragdoll_LevelSwitchDoesNotCrash/NoRagdoll 
C13895144_Ragdoll_WithRagdoll                                  ->  Ragdoll_LevelSwitchDoesNotCrash/WithRagdoll 
Ragdoll_MutipleRagdollsAndLevelSwitchDoesNotCrash              ->  Ragdoll_LevelSwitchDoesNotCrash 
C14654882_Ragdoll_ragdollAPTest                                ->  Ragdoll_OldRagdollSerializationNoErrors 
C17411467_AddPhysxRagdollComponent                             ->  Ragdoll_AddPhysxRagdollComponentWorks 
C28978033_Ragdoll_WorldBodyBusTests                            ->  Ragdoll_WorldBodyBusWorks 
C100000_RigidBody_EnablingGravityWorksPoC                      ->  RigidBody_EnablingGravityWorksPoC 
C111111_RigidBody_EnablingGravityWorksUsingNotificationsPoC    ->  RigidBody_EnablingGravityWorksUsingNotificationsPoC 
C4976194_RigidBody_PhysXComponentIsValid                       ->  RigidBody_AddRigidBodyComponent 
C4976195_RigidBodies_InitialLinearVelocity                     ->  RigidBody_InitialLinearVelocity 
C4976197_RigidBodies_InitialAngularVelocity                    ->  RigidBody_InitialAngularVelocity 
C4976199_RigidBodies_LinearDampingObjectMotion                 ->  RigidBody_LinearDampingAffectsMotion 
C4976200_RigidBody_AngularDampingObjectRotation                ->  RigidBody_AngularDampingAffectsRotation 
C4976201_RigidBody_MassIsAssigned                              ->  RigidBody_MassDifferentValuesWorks 
C4976202_RigidBody_StopsWhenBelowKineticThreshold              ->  RigidBody_SleepWhenBelowKineticThreshold 
C4976204_Verify_Start_Asleep_Condition                         ->  RigidBody_StartAsleepWorks 
C4976206_RigidBodies_GravityEnabledActive                      ->  RigidBody_StartGravityEnabledWorks 
C4976206_RigidBodies_VerifyGravity                             ->  RigidBody_StartGravityEnabledWorks 
C4976207_PhysXRigidBodies_KinematicBehavior                    ->  RigidBody_KinematicModeWorks 
C4976209_RigidBody_ComputesCOM                                 ->  RigidBody_COM_ComputingWorks 
C4976210_COM_ManualSetting                                     ->  RigidBody_COM_ManualSettingWorks 
C4976218_RigidBodies_InertiaObjectsNotComputed                 ->  RigidBody_ComputeInertiaWorks 
C4976218_RigidBodies_InertiaObjectNotComputed                  ->  RigidBody_ComputeInertiaWorks 
C5340400_RigidBody_ManualMomentOfInertia                       ->  RigidBody_MomentOfInertiaManualSetting 
C13351703_COM_NotIncludeTriggerShapes                          ->  RigidBody_COM_NotIncludesTriggerShapes 
C13352089_RigidBodies_MaxAngularVelocity                       ->  RigidBody_MaxAngularVelocityWorks 
C6224408_ScriptCanvas_EntitySpawn                              ->  ScriptCanvas_SpawnEntityWithPhysComponents 
C6274125_ScriptCanvas_TriggerEvents                            ->  ScriptCanvas_TriggerEvents 
C12712452_ScriptCanvas_CollisionEvents                         ->  ScriptCanvas_CollisionEvents 
C12712453_ScriptCanvas_MultipleRaycastNode                     ->  ScriptCanvas_MultipleRaycastNode 
C12712454_ScriptCanvas_OverlapNodeVerification                 ->  ScriptCanvas_OverlapNode 
C12712455_ScriptCanvas_ShapeCastVerification                   ->  ScriptCanvas_ShapeCast 
C14195074_ScriptCanvas_PostUpdateEvent                         ->  ScriptCanvas_PostUpdateEvent 
C14902097_ScriptCanvas_PreUpdateEvent                          ->  ScriptCanvas_PreUpdateEvent 
C14902098_ScriptCanvas_PostPhysicsUpdate                       ->  ScriptCanvas_PostPhysicsUpdate 
C14976308_ScriptCanvas_SetKinematicTargetTransform             ->  ScriptCanvas_SetKinematicTargetTransform 
C3510642_Terrain_NotCollideWithTerrain                         ->  Terrain_NoPhysTerrainComponentNoCollision 
C5689518_PhysXTerrain_CollidesWithPhysXTerrain                 ->  Terrain_CollisionAgainstRigidBody 
C5689518_PhysXEntity_collides_with_PhysXTerrain                ->  Terrain_CollisionAgainstRigidBody 
C5689522_Physxterrain_AddPhysxterrainNoEditorCrash             ->  Terrain_AddPhysTerrainComponent 
C5689524_MultipleTerrains_CheckWarningInConsole                ->  Terrain_CanAddMultipleTerrainComponents 
C5689528_Terrain_MultipleTerrainComponents                     ->  Terrain_MultipleTerrainComponentsWarning 
C5689531_Warning_TerrainSliceTerrainComponent                  ->  Terrain_SpawnSecondTerrainComponentWarning 
C6032082_Terrain_MultipleResolutionsValid                      ->  Terrain_MultipleResolutionsValid 
C6032082_Terrain_MultipleResolutionsValid_1024x1024_32m        ->  Terrain_MultipleResolutionsValid/1024x1024_32m 
C6032082_Terrain_MultipleResolutionsValid_128x128_1m           ->  Terrain_MultipleResolutionsValid/128x128_1m 
C6032082_Terrain_MultipleResolutionsValid_4096x4096_16m        ->  Terrain_MultipleResolutionsValid/4096x4096_16m 
C6032082_Terrain_MultipleResolutionsValid_512x512_2m           ->  Terrain_MultipleResolutionsValid/512x512_2m 
C13508019_Terrain_TerrainTexturePainterWorks                   ->  Terrain_TerrainTexturePainterWorks 
C15308217_NoCrash_LevelSwitch                                  ->  Terrain_TextureLayerToolNoCrashOnLevelChange 
c15096732_material_defaultlibraryupdatedacrosslevels           ->  c15096732_material_defaultlibraryupdatedacrosslevels
```
NvCloth renames:
```
C18977329_NvCloth_AddClothSimulationToMesh -> NvCloth_AddClothSimulationToMesh
C18977330_NvCloth_AddClothSimulationToActor -> NvCloth_AddClothSimulationToActor
```

WhiteBox renames:
```
C28798177_WhiteBox_AddComponentToEntity -> WhiteBox_AddComponentToEntity(
C29279329_WhiteBox_SetDefaultShape -> WhiteBox_SetDefaultShape
C28798205_WhiteBox_SetInvisible -> WhiteBox_SetInvisible
```

Blast renames:
```
ActorSplitsAfterCollision -> Blast_ActorSplitsAfterCollision
ActorSplitsAfterRadialDamage -> Blast_ActorSplitsAfterRadialDamage
ActorSplitsAfterCapsuleDamage -> Blast_ActorSplitsAfterCapsuleDamage
ActorSplitsAfterImpactSpreadDamage -> Blast_ActorSplitsAfterImpactSpreadDamage
ActorSplitsAfterShearDamage -> Blast_ActorSplitsAfterShearDamage
ActorSplitsAfterTriangleDamage -> Blast_ActorSplitsAfterTriangleDamage
ActorSplitsAfterStressDamage -> Blast_ActorSplitsAfterStressDamage
```

Extra changes:
- Fixed the console output window in batchmode so it shows black text when there is white background
- Fixed bug in optimized test suite to use the correct paths when doing file file overrides on different paths
- Fixed Default material library test
- Moved Joints tests out of InDevelopment into Periodic

### CODE REVIEW

**Comment by AMZN-AlexOteiza** [comment]
Sorry for the big amount of files, this is due to the lots of renames and file relocations.
I separated this review in multiple commits, for easing the review, you do changes per commit:
image

File: AutomatedTesting/Gem/PythonTests/physics/TestSuite_Main_Optimized.py [hunk 1]
```diff
@@ -47,14 +47,14 @@ def wrap_run(cls, instance, request, workspace, editor, editor_test_results, lau
...
-            fm._copy_file(override, file_list[override], original, file_list[override])
+            fm._copy_file(override, file_list[override], original, file_list[original])
 
         yield # Run Test
         for f in original_file_list:
             fm._restore_file(f, file_list[f])
 
 
-@pytest.mark.xfail(reason="Optimized tests are experimental, we will enable xfail and monitor them temporarily.")
+#@pytest.mark.xfail(reason="Optimized tests are experimental, we will enable xfail and monitor them temporarily.")
```
Comment (AMZN-AlexOteiza) [inline AutomatedTesting/Gem/PythonTests/physics/TestSuite_Main_Optimized.py:57]: I need to remove this #, it was for local testing

**Review by hultonha — APPROVED** [review]
I understand why doing this in one go is convenient and useful but it does make reviews incredibly difficult. It's much easier if these changes are done incrementally in smaller batches. This can add additional overhead but it's a lot safer and easier to spot issues and catch regressions. I'll approve as I think holding this up now is going to cause more harm than good, but I would strongly advice against this approach in future.

**Review by moraaar — APPROVED** [review]
I took  a quick look, approving to move forward as it's very difficult to review. Please make sure you run all the test suites both locally and through Jenkins. The results should be the same as in both gating AR and nightly builds.

**Pull request merged**